packages/backend-api/src/Component/DataSetter/ApiDataSetter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\BackendApiBundle\Component\DataSetter;
+
+use DateTime;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+
+/**
+ * @experimental
+ */
+class ApiDataSetter
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(Domain $domain)
+    {
+        $this->domain = $domain;
+    }
+
+    /**
+     * @param string $arrayKey
+     * @param array $apiData
+     * @param object $entityData
+     * @param string|null $entityDataKey
+     */
+    public function setValueIfExists(string $arrayKey, array $apiData, object $entityData, ?string $entityDataKey = null): void
```

Comment:
`object` is really wide. I can pass any object I want? 
Couldn't it be narrowed?

Reply:
Well idea of this `ApiDataSetter` is to be able to set any data object in the future. So it is true that right now it sets only `ProductData`, but in the future it could be `OrderData`, ...

So I can't come with a better typehint

Reply:
Well the interface came in my mind, but I have some feeling that you will not agree 😄

Reply:
correct

Reply:
okay then :)